src/glum/_transformers.py

```diff
@@ -0,0 +1,347 @@
+"""Some utilities for transforming data before fitting a model."""
+
+from typing import Hashable, List, NamedTuple, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import tabmat as tm
+from scipy.linalg import qr
+from sklearn.base import BaseEstimator, TransformerMixin, check_is_fitted
+
+from ._glm import ArrayLike, VectorLike
+from ._util import _safe_sandwich_dot
+
+
+class CollinearityResults(NamedTuple):
+    """Results of collinearity analysis."""
+
+    keep_idx: Sequence[int]
+    drop_idx: Sequence[int]
+    intercept_safe: bool
+
+
+def _find_collinear_columns_pandas(
+    df: pd.DataFrame,
+    fit_intercept=True,
+    mode: str = "gram",
+    use_tabmat: bool = True,
+    tolerance=1e-6,
+) -> CollinearityResults:
+    """Decompose a pandas.DataFrame into a tabmat.SplitMatrix.
```

Comment:
Does not seem to be the main purpose of this function.

Reply:
True, fixed